Say Their Names/Source/Controller/Home/CollectionViewDataSource/LocationCollectionViewDataSource.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  LocationCollectionViewDataSource.swift
+//  Say Their Names
+//
+//  Created by JohnAnthony on 6/2/20.
+//  Copyright © 2020 Franck-Stephane Ndame Mpouli. All rights reserved.
+//
+
+import UIKit
+
+class LocationCollectionViewDataSource: NSObject {
+     
+    private var locations: [Location]
+    
+    init(locations: [Location]) {
+        self.locations = locations
+    }
+    
+    func setLocations(_ locations: [Location]) {
+        self.locations = locations
+    }
+    
+}
+
+extension LocationCollectionViewDataSource: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        locations.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+         
+        let location = locations[indexPath.item]
+        
+        if let locationCell = collectionView.dequeueReusableCell(withReuseIdentifier: LocationCell.locationIdentifier, for: indexPath) as? LocationCell {
```

Comment:
We have an extension that can clean this up if we make `LocationCell` conform to `Reusable`
```suggestion
        let locationCell: LocationCell = collectionView.dequeueCell(for: indexPath)
```